### ISSUE TITLE
Make TimeOfDay comparable

### ISSUE THREAD

**Issue by mj1856** [mj1856/corefx-dateandtime#30]
From a strict modeling perspective, it doesn't make sense for `TimeOfDay` to be comparable when you think about time on a 24 hour clock.  Since two hours after 23:00 is 01:00, then who's to say that 23:00 is the greater value?

However, in the real world, I think that there will be many scenarios that will require comparison.  For example, expressing a LINQ or OData query.  For these usages, it will not be easy to use the `TimeOfDay.IsBetween` method.

Also, the general intention is that `TimeOfDay` would map closely to SQL Server's `time` data type, which is comparable in the context of the `WHERE` or `ORDER BY` clause of a SQL query.

In other words, `TimeOfDay` should be comparable, but when comparing it should be expressly comparing the *magnitude* of the time, rather than its position on a clock.

- `t1 < t2` would be equivalent to `t1.IsBetween(new TimeOfDay(0,0), t2)`
- `t1 > t2` would be equivalent to `t1.IsBetween(t2, new TimeOfDay(0,0))`

**Comment by mj1856** [comment]
Completed in 24324f1f0970bf75abf9701eb3dde4db02e2a574
